OpenBSD-7.9 smoker reporting failure at onset of 'make'

We frequently get smoke-test reports from a rig set up on OpenBSD-6.9 at host perl-sparc64-stable.home.hewus.com.  This host has been reporting build-time failures since approximately Aug 05 2026.  The first failing test report can be viewed [here](https://perl5.test-smoke.org/report/5536303).  Excerpt:
```
Automated smoke report for branch blead 5.45.2 patch 703fd1abb1725d86150c941d8c157afdb27c0160 v5.45.1-132-g703fd1abb1.
perl-sparc64-stable.home.hewus.com: SPARC-T4 (rev 0.0) @ 2847.862 MHz (2847 MHz) (sparc64/12 cpus)
    on        openbsd 7.9
    using     cc version 4.2.1 20070719 (*1)
    smoketime 30 minutes 47 seconds (average 3 minutes 50.875 seconds)

Summary: FAIL(m)

O = OK  F = Failure(s), extended report at the bottom
X = Failure(s) under TEST but not under harness
? = still running or test results not (yet) available
Build failures during:       - = unknown or N/A
c = Configure, m = make, M = make (after miniperl), t = make test-prep

v5.45.1-132-g703fd1abb1
----------------------- ----------------------------------------------------
m  m                     (*1)
m  m                    -Duseithreads (*1)
m  m                    -Acppflags=-DBIG_TIME -DNO_LOCALE_NUMERIC -DNO_LOCALE_COLLATE (*1)
m  m                    -Acppflags=-DBIG_TIME -DNO_LOCALE_NUMERIC -DNO_LOCALE_COLLATE -Duseithreads (*1)
|  |
|  +------------------  stdio DEBUGGING
+---------------------  stdio
```
In the corresponding [log file](https://perl.develop-help.com/dblog/5536303), the only diagnostic is `Unable to make perl in this configuration`.

I contacted @afresh1 (report submitter), who went to the machine in question, configured and attempted to run `make`.  He got this error immediately:
```
perl-sparc64-stable$ make
echo @`sh  cflags "optimize='-O2'" opmini.o`  -DPERL_IS_MINIPERL -DPERL_EXTERNAL_GLOB opmini.c
@cc -c -DPERL_CORE -DNO_LOCALE_NUMERIC -DNO_LOCALE_COLLATE -DNO_LOCALE_MONETARY -DNO_LOCALE_TIME -DNO_LOCALE_MESSAGES -DLIBC_HANDLES_MISMATC
HED_CTYPE -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/local/include -D_FORTIFY_SOURCE=2 -std=c99 -O2 -Wall -Werror=pointer-ar
ith -Wextra -Wno-long-long -Wno-declaration-after-statement -Wno-nonnull-compare -Wc++-compat -Wno-c++-keyword -Wwrite-strings -DPERL_IS_MINIPERL -DPERL_EXTERNAL_GLOB opmini.c
[...]
At top level:
cc1: error: unrecognized command line option "-Wno-c++-keyword"
*** Error 1 in /home/smoker/perl-test (makefile:313 'opmini.o': @`sh  cflags "optimize='-O2'" opmini.o`  -DPERL_IS_MINIPERL -DPERL_EXTERNAL_...)
```
He eventually ran a bisection, which pointed to this commit:
```
commit aca7b5cadb252ed5eb4a57fe2a8980ce24725651 (HEAD)
Author: Paul Evans <leonerd@leonerd.org.uk>
Date:   Wed Apr 15 21:46:36 2026 +0100
Commit:     Paul Evans <leonerd@leonerd.org.uk>
CommitDate: Tue Aug 4 12:27:37 2026 +0100

    Create concept of SV-backed PADNAMEs
    
    Currently entirely untested. Could do with some actual use, especially
    during thread cloning, outer captures, and eventual reclaim afterwards.
```
That's consistent with the date of the first FAIL report.  @afresh1 also reported more error output:
```
echo @`sh  cflags "optimize='-O2'" opmini.o`  -DPERL_IS_MINIPERL -DPERL_EXTERNAL_GLOB opmini.c
@cc -c -DPERL_CORE -DNO_LOCALE_NUMERIC -DNO_LOCALE_COLLATE -DNO_LOCALE_MONETARY -DNO_LOCALE_TIME -DNO_LOCALE_MESSAGES -DLIBC_HANDLES_MISMATCHED_CTYPE -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/local/include -D_FORTIFY_SOURCE=2 -std=c99 -O2 -Wall -Werror=pointer-arith -Wextra -Wno-long-long -Wno-declaration-after-statement -Wno-nonnull-compare -Wc++-compat -Wno-c++-keyword -Wwrite-strings -DPERL_IS_MINIPERL -DPERL_EXTERNAL_GLOB opmini.c
In file included from perl.h:4593,
                 from op.c:163:
pad.h:79: warning: declaration does not declare anything
pad.h:84: warning: declaration does not declare anything
op.c: In function 'Perl_check_hash_fields_and_hekify':
op.c:2873: error: 'PADNAME' has no member named 'xpadn_sv'
op.c:2873: error: 'PADNAME' has no member named 'xpadn_sv'
op.c:2873: error: 'PADNAME' has no member named 'xpadn_pv'
op.c: In function 'Perl_op_lvalue_flags':
op.c:3469: error: 'PADNAME' has no member named 'xpadn_sv'
op.c:3469: error: 'PADNAME' has no member named 'xpadn_sv'
op.c:3469: error: 'PADNAME' has no member named 'xpadn_pv'
op.c: In function 'walk_ops_forbid':
op.c:5654: warning: value computed is not used
op.c: In function 'S_set_haseval':
op.c:8014: error: 'PADNAME' has no member named 'xpadn_sv'
op.c: In function 'Perl_newFOROP':
op.c:10540: error: 'PADNAME' has no member named 'xpadn_sv'
op.c:10540: error: 'PADNAME' has no member named 'xpadn_pv'
op.c:10542: error: 'PADNAME' has no member named 'xpadn_sv'
op.c: In function 'S_already_defined':
op.c:11265: error: 'PADNAME' has no member named 'xpadn_sv'
op.c:11265: error: 'PADNAME' has no member named 'xpadn_pv'
op.c:11265: error: 'PADNAME' has no member named 'xpadn_sv'
op.c: In function 'Perl_newMYSUB':
op.c:11342: error: 'PADNAME' has no member named 'xpadn_sv'
op.c:11342: error: 'PADNAME' has no member named 'xpadn_sv'
op.c:11342: error: 'PADNAME' has no member named 'xpadn_pv'
op.c:11342: error: 'PADNAME' has no member named 'xpadn_sv'
op.c:11376: error: 'PADNAME' has no member named 'xpadn_sv'
op.c:11376: error: 'PADNAME' has no member named 'xpadn_sv'
op.c:11376: error: 'PADNAME' has no member named 'xpadn_pv'
op.c:11376: error: 'PADNAME' has no member named 'xpadn_sv'
op.c:11376: error: 'PADNAME' has no member named 'xpadn_sv'
op.c:11376: error: 'PADNAME' has no member named 'xpadn_pv'
op.c:11376: error: 'PADNAME' has no member named 'xpadn_sv'
op.c:11377: error: 'PADNAME' has no member named 'xpadn_sv'
op.c:11377: error: 'PADNAME' has no member named 'xpadn_pv'
op.c:11377: error: 'PADNAME' has no member named 'xpadn_sv'
op.c:11423: error: 'PADNAME' has no member named 'xpadn_sv'
op.c:11423: error: 'PADNAME' has no member named 'xpadn_sv'
op.c:11423: error: 'PADNAME' has no member named 'xpadn_pv'
op.c:11423: error: 'PADNAME' has no member named 'xpadn_sv'
op.c:11537: error: 'PADNAME' has no member named 'xpadn_sv'
op.c:11537: error: 'PADNAME' has no member named 'xpadn_sv'
op.c:11537: error: 'PADNAME' has no member named 'xpadn_pv'
op.c:11537: error: 'PADNAME' has no member named 'xpadn_sv'
op.c:11537: error: 'PADNAME' has no member named 'xpadn_sv'
op.c:11537: error: 'PADNAME' has no member named 'xpadn_pv'
op.c:11537: error: 'PADNAME' has no member named 'xpadn_sv'
op.c:11538: error: 'PADNAME' has no member named 'xpadn_sv'
op.c:11538: error: 'PADNAME' has no member named 'xpadn_pv'
op.c:11538: error: 'PADNAME' has no member named 'xpadn_sv'
op.c:11600: error: 'PADNAME' has no member named 'xpadn_sv'
op.c:11600: error: 'PADNAME' has no member named 'xpadn_pv'
op.c:11600: error: 'PADNAME' has no member named 'xpadn_sv'
op.c: In function 'Perl_ck_eval':
op.c:13632: warning: value computed is not used
op.c: In function 'Perl_ck_fun':
op.c:14085: error: 'PADNAME' has no member named 'xpadn_sv'
op.c:14085: error: 'PADNAME' has no member named 'xpadn_pv'
op.c:14086: error: 'PADNAME' has no member named 'xpadn_sv'
op.c:14124: error: 'PADNAME' has no member named 'xpadn_sv'
op.c:14124: error: 'PADNAME' has no member named 'xpadn_pv'
op.c: In function 'S_simplify_sort':
op.c:15169: error: 'PADNAME' has no member named 'xpadn_sv'
op.c:15169: error: 'PADNAME' has no member named 'xpadn_sv'
op.c:15169: error: 'PADNAME' has no member named 'xpadn_pv'
op.c:15170: error: 'PADNAME' has no member named 'xpadn_sv'
op.c:15170: error: 'PADNAME' has no member named 'xpadn_pv'
op.c:15171: error: 'PADNAME' has no member named 'xpadn_sv'
op.c:15171: error: 'PADNAME' has no member named 'xpadn_pv'
op.c:15173: error: 'PADNAME' has no member named 'xpadn_sv'
op.c:15173: error: 'PADNAME' has no member named 'xpadn_pv'
op.c: In function 'Perl_find_lexical_cv':
op.c:15437: error: 'PADNAME' has no member named 'xpadn_sv'
op.c:15437: error: 'PADNAME' has no member named 'xpadn_sv'
op.c:15438: error: 'PADNAME' has no member named 'xpadn_sv'
op.c:15438: error: 'PADNAME' has no member named 'xpadn_pv'
op.c:15438: error: 'PADNAME' has no member named 'xpadn_sv'
op.c:15438: error: 'PADNAME' has no member named 'xpadn_pv'
At top level:
cc1: error: unrecognized command line option "-Wno-c++-keyword"
*** Error 1 in /home/afresh1/perl5 (makefile:313 'opmini.o': @`sh  cflags "optimize='-O2'" opmini.o`  -DPERL_IS_MINIPERL -DPERL_EXTERNAL_GLO...)
```
On irc.perl.org, @afresh1 hypothesized that this could be due to the very old C-compiler on this machine: gcc-4.  He's continuing to investigate.  It should be noted that we are getting PASSing smoke-test [reports](https://perl5.test-smoke.org/report/5537564) from @cjg-cguevara, also on OpenBSD-7.9, where the C-compiler in use is gcc-15.

@leonerd, can you take a look?  (Contact @afresh1 for login.)  Thanks.

--- a/pad.c
+++ b/pad.c
@@ -2951,7 +2951,7 @@ Perl_padname_upgrade_sv(pTHX_ PADNAME *pn)
     SvUTF8_on(sv);
     SvREADONLY_on(sv);
 
-    pn->xpadn_sv = sv;
+    pn->xpadn_name_u.xpadn_sv = sv;
 
     PadnameFLAGS(pn) |= PADNAMEf_FULLSV;
     return pn;
@@ -2995,8 +2995,8 @@ Perl_padname_dup(pTHX_ PADNAME *src, CLONE_PARAMS *param)
 
         if(PadnameIsFULLSV(src)) {
             char *pnpv = PadnamePV(dst);
-            dst->xpadn_sv = sv_dup_inc(src->xpadn_sv, param);
-            SvPVX(dst->xpadn_sv) = pnpv;
+            dst->xpadn_name_u.xpadn_sv = sv_dup_inc(src->xpadn_name_u.xpadn_sv, param);
+            SvPVX(dst->xpadn_name_u.xpadn_sv) = pnpv;
         }
     }
     ptr_table_store(PL_ptr_table, src, dst);

--- a/pad.h
+++ b/pad.h
@@ -61,7 +61,7 @@ struct padname_fieldinfo;
     union {                             \
         char *	xpadn_pv;		\
         SV *    xpadn_sv;               \
-    };                                  \
+    } xpadn_name_u;                     \
     HV *	xpadn_ourstash;		\
     union {				\
         HV *	xpadn_typestash;	\
@@ -345,11 +345,11 @@ Restore the old pad saved into the local variable C<opad> by C<PAD_SAVE_LOCAL()>
 #define PadARRAY(pad)		AvARRAY(pad)
 #define PadMAX(pad)		AvFILLp(pad)
 
-#define PadnamePV(pn)		(PadnameIsFULLSV(pn) ? SvPVX((pn)->xpadn_sv) : (pn)->xpadn_pv)
-#define PadnameLEN(pn)		(PadnameIsFULLSV(pn) ? SvCUR((pn)->xpadn_sv) : (pn)->xpadn_len)
+#define PadnamePV(pn)		(PadnameIsFULLSV(pn) ? SvPVX((pn)->xpadn_name_u.xpadn_sv) : (pn)->xpadn_name_u.xpadn_pv)
+#define PadnameLEN(pn)		(PadnameIsFULLSV(pn) ? SvCUR((pn)->xpadn_name_u.xpadn_sv) : (pn)->xpadn_len)
 #define PadnameUTF8(pn)		1
 #define PadnameSV(pn) \
-        (PadnameIsFULLSV(pn) ? (pn)->xpadn_sv : newSVpvn_flags(PadnamePV(pn), PadnameLEN(pn), SVs_TEMP|SVf_UTF8|SVf_READONLY|SVf_PROTECT))
+        (PadnameIsFULLSV(pn) ? (pn)->xpadn_name_u.xpadn_sv : newSVpvn_flags(PadnamePV(pn), PadnameLEN(pn), SVs_TEMP|SVf_UTF8|SVf_READONLY|SVf_PROTECT))
 #define PadnameFLAGS(pn)	(pn)->xpadn_flags
 #define PadnameIsOUR(pn)	cBOOL((pn)->xpadn_ourstash)
 #define PadnameOURSTASH(pn)	(pn)->xpadn_ourstash
@@ -380,7 +380,7 @@ Restore the old pad saved into the local variable C<opad> by C<PAD_SAVE_LOCAL()>
 #define PADNAMEf_FULLSV     0x80    /* padname stored in SVt_PV in xpadn_sv */
 
 #ifdef PERL_CORE
-#  define PadnamePV_set(pn, pv)     (assert(!PadnameIsFULLSV(pn)), (pn)->xpadn_pv = (pv))
+#  define PadnamePV_set(pn, pv)     (assert(!PadnameIsFULLSV(pn)), (pn)->xpadn_name_u.xpadn_pv = (pv))
 #  define PadnameLEN_set(pn, len)   (assert(!PadnameIsFULLSV(pn)), (pn)->xpadn_len = (len))
 #endif
 

